### PR TITLE
feat: add version filter regex/semver

### PR DESCRIPTION
**Pull request by mcwarman**
I've had a couple of cases where i wanted to get latest semantic version, but the versions returned by the resource do not follow the convention.

The idea behind this to extract the version numbers using a regex then pass them on to the semver search.

[yarnpkg/berry](https://github.com/yarnpkg/berry) is a good example, as they do backport releases. Therefore latest of both `regex` and `latest` currently returns a `3.x`.

## Test

To test this pull request, you can run the following commands:

```shell
cd pkg/plugins/utils/version
go test
```

```shell
go build -o bin/updatecli
./bin/updatecli diff --config test.yaml
```

```yaml
sources:
  default:
    name: Get latest version
    kind: githubrelease
    spec:
      owner: yarnpkg
      repository: berry
      token: '{{ requiredEnv "GITHUB_TOKEN" }}'
      versionfilter:
        kind: regex/semver
        regex: "@yarnpkg/cli/(\\d*\\.\\d*\\.\\d*)"
        #pattern: "3.x"
```

### Tradeoff

Potentially confusing having `regex` parameter.

### Potential improvement

N/A

### CODE REVIEW

File: pkg/plugins/utils/version/filter.go [hunk 1]
```diff
@@ -17,16 +17,17 @@ const (
 	SEMVERVERSIONKIND string = "semver"
 	// LATESTVERSIONKIND specifies that we are looking for the latest version of an array
 	LATESTVERSIONKIND string = "latest"
+	// REGEXSEMVERVERSIONKIND use regex to extract versions and semantic version to find version
+	REGEXSEMVERVERSIONKIND string = "regex-semver"
```
Comment (mcwarman) [inline pkg/plugins/utils/version/filter.go:21]: I made this kebab case, this might not follow convention though.

Comment (olblak) [reply pkg/plugins/utils/version/filter.go:21]: To align with other plugin as you can see [here](https://github.com/updatecli/updatecli/blob/caa6cef9fe86cf172abe53a8e2c8484b2d81ed5c/pkg/core/pipeline/resource/main.go#L86), I would use "regex/semver"

**Comment by olblak** [comment]
I like this pullr equest a lot, because I faced the same issues.
Like the kubernetes publish version with "kubernetes-" prefix

File: pkg/plugins/utils/version/filter.go (hunk 1 above)
Comment (mcwarman) [reply pkg/plugins/utils/version/filter.go]: Perfect, it didn't feel right, and I didn't really engage my brain to resolve it.

**Comment by mcwarman** [comment]
@olblak not sure on test failures seems unrelated.

**Review by olblak — APPROVED** [review]
Thanks for the pull request

**Pull request merged**